src/VisualStudio/Core/Def/Diagnostics/VisualStudioDiagnosticAnalyzerService.cs

```diff
@@ -424,27 +432,35 @@ private void UpdateStatusCore()
                 await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync();
 
                 string message;
-                int fInProgress;
+                bool inProgress;
                 var analyzedProjectCount = (uint)_analyzedProjectCount;
                 if (analyzedProjectCount == _totalProjectCount)
                 {
-                    message = _statusMesageOnCompleted;
-                    fInProgress = 0;
+                    message = _statusMessageOnCompleted;
+                    inProgress = false;
                 }
                 else if (_disposed)
                 {
-                    message = _statusMesageOnTerminated;
-                    fInProgress = 0;
+                    message = _statusMessageOnTerminated;
+                    inProgress = false;
                 }
                 else
                 {
                     message = _statusMessageWhileRunning;
-                    fInProgress = 1;
+                    inProgress = true;
                 }
 
-                // Update the status bar progress and text.
-                _statusBar.Progress(ref _statusBarCookie, fInProgress, message, analyzedProjectCount, _totalProjectCount);
-                _statusBar.SetText(message);
+                // Update the task progress and text.
+                _taskHandler.Progress.Report(
+                    new TaskProgressData()
+                    {
+                        CanBeCanceled = inProgress,
+                        PercentComplete = _totalProjectCount > 0 ? (int)Math.Round(100 * (double)analyzedProjectCount / _totalProjectCount) : null,
+                        ProgressText = message,
+                    });
+
+                if (!inProgress)
+                    _taskCompletionSource.SetResult(true);
             });
```

Comment:
should this be passed the CT?  if not, can you doc that we are intentionally not canceling this